include/cantera/kinetics/ReactionRate.h

```diff
@@ -216,6 +216,35 @@ class ReactionRate
         m_composition_dependent_rate = comp_dep;
     }
 
+    //! @name Properties of the electron-collision-plasma rate
+    //! @{
+
+    //! The value of #ElectronCollisionPlasmaRate::m_energyLevels [eV]
+    virtual const vector<double>& energyLevels() const {
+        throw NotImplementedError("ReactionRate::energyLevels",
+                                  "Not implemented by '{}' object.", type());
+    }
```

Comment:
Can we avoid introducing these as virtual methods on the base `ReactionRate` class, given they're only used in one specific instance? I think you'd just want to change `PlasmaPhase::m_collisions` to hold pointers to the derived type and use `dynamic_pointer_cast` to get them from the `shared_ptr<ReactionRate>` objects.

Reply:
I use m_collisionRates to hold ElectronCollisionPlasmaRate to prevent dynamic casting.